x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/ActionConfigStatsTests.java

```diff
@@ -96,7 +96,12 @@ protected ActionConfigStats mutateInstance(ActionConfigStats instance) throws IO
                 builder.setShrinkMaxPrimaryShardSize(randomPrimaryByteSize);
                 break;
             case 8:
-                builder.setShrinkNumberOfShards(randomIntBetween(0, 50));
+                Integer oldValue = instance.getShrinkNumberOfShards();
+                int newValue;
+                do {
+                    newValue = randomIntBetween(0, 50);
+                } while (oldValue != null && newValue == oldValue);
+                builder.setShrinkNumberOfShards(newValue);
```

Comment:
```suggestion
                builder.setShrinkNumberOfShards(randomValueOtherThan(instance.getShrinkNumberOfShards(), () -> randomIntBetween(0, 50)));
```

Reply:
Thanks for this, I was looking for this method but somehow missed it